uobtheatre/payments/transferables.py

```diff
@@ -0,0 +1,91 @@
+import math
+from abc import abstractmethod
+from typing import TYPE_CHECKING, Optional
+
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+
+from uobtheatre.payments.models import SalesBreakdown, Transaction
+from uobtheatre.payments.payables import Payable
+
+if TYPE_CHECKING:
+    pass
+
+# pylint: disable=abstract-method
+class Transferable(Payable):
+    """
+    A transferable is a payable which can be transfered. A transfer is a
+    to/from another transferable of the same type. For bookings this enables bookings to be updated (chnaged perfomrance)
+    """
+
+    # Whether the booking has be transferred to
+    transfered_from = models.OneToOneField(
+        "self", on_delete=models.RESTRICT, null=True, related_name="_transfered_to"
+    )
+
+    @property
+    def transfered_to(self) -> Optional["Transferable"]:
+        """The transfered which this has been transfred to.
+
+        If this has not be transfered None will be returned.
+
+        Note: _transfered_to cannot be used directly as if this booking has not
+        been transfered an error is raised.
+        """
+        if not hasattr(self, "_transfered_to"):
+            return None
+        return self._transfered_to  # type: ignore
+
+    @property
+    def transfered_from_chain(self) -> list["Transferable"]:
+        """
+        A booking can be transfered which creates a new booking and cancels the
+        existing one. If this booking has been created in a transfer, this
+        gives all bookings which this has been transfered from. Multiple are
+        possible as a booking which is created in a transfer can it self be
+        transfered.
+
+        Returns:
+            list[int]: All the bookings in this transfer chain
+        """
+        if not self.transfered_from:
+            return []
+        return [self.transfered_from] + self.transfered_from.transfered_from_chain
+
+    @property
+    def transfer_reduction(self) -> int:
+        """
+        When transfering a payable, all previous payments (excluding
+        transfer fees) are excluded.
```

Comment:
This comment doesn't actually explain what comes out of this property. By the looks of it, it is the amount of money the customer has already paid for preceding bookings. Not sure `transfer_reduction` is the best phrasing. Maybe `total_transfer_chain_payments`? Appreciate that it excludes transfer fees, but something doesn't sit right about the name